### DISCUSSION TITLE
Request with already cancelled CancelToken

### DISCUSSION THREAD

**Discussion by AndreiMisiukevich** [cfug/dio#2207] [Q&A]
Does a server request still get initiated if an already-cancelled CancelToken is passed along with the request? ```cancelToken.isCancelled // true```

I've briefly looked through the code and didn't find any places like that
```if (cancelToken?.cancelError != null) throw cancelToken!.cancelError!;```

And if my assumption is correct, a server request always gets initiated. And then immediately gets cut off.

**Comment by AlexV525** [comment]
Implemented in #2251.
